kakao/src/main/kotlin/com/agoda/kakao/configurator/DataInteractionDelegateBuilder.kt

```diff
@@ -0,0 +1,49 @@
+package com.agoda.kakao.configurator
+
+import android.support.test.espresso.DataInteraction
+import android.support.test.espresso.ViewAssertion
+import android.support.test.espresso.ViewInteraction
+import android.view.View
+import com.agoda.kakao.delegates.DataInteractionDelegate
+import com.agoda.kakao.delegates.ViewInteractionDelegate
+import com.agoda.kakao.delegates.factory.InteractionDelegatesFactory
+import org.hamcrest.Matcher
+
+class DataInteractionDelegateBuilder(
```

Comment:
Why do we need builders for delegates? I think delegates should be final classes, with a stack of interceptors

Reply:
Have you ever seen Kaspresso interceptors' concept? I can add you to the project.
I think it's a bad idea to mix both concepts. 
Kakao - is for delegates.
Kaspresso - one of 3rd party libs that implements Kakao delegates with Interceptors approach.